power_grid_model_c/power_grid_model_cpp/include/power_grid_model_cpp/options.hpp

```diff
@@ -0,0 +1,78 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_CPP_OPTIONS_HPP
+#define POWER_GRID_MODEL_CPP_OPTIONS_HPP
+
+#include "basics.hpp"
+#include "handle.hpp"
+
+#include "power_grid_model_c/options.h"
+
+namespace power_grid_model_cpp {
+class Options {
+  public:
+    Options() : options_{handle_.call_with(PGM_create_options)} {}
+
+    RawOptions* get() const { return options_.get(); }
+
+    static void set_calculation_type(Options& options, Idx type) {
+        options.handle_.call_with(PGM_set_calculation_type, options.get(), type);
+    }
+    void set_calculation_type(Idx type) { set_calculation_type(*this, type); }
+
+    static void set_calculation_method(Options& options, Idx method) {
+        options.handle_.call_with(PGM_set_calculation_method, options.get(), method);
+    }
+    void set_calculation_method(Idx method) { set_calculation_method(*this, method); }
+
+    static void set_symmetric(Options& options, Idx sym) {
+        options.handle_.call_with(PGM_set_symmetric, options.get(), sym);
+    }
+    void set_symmetric(Idx sym) { set_symmetric(*this, sym); }
+
+    static void set_err_tol(Options& options, double err_tol) {
+        options.handle_.call_with(PGM_set_err_tol, options.get(), err_tol);
+    }
+    void set_err_tol(double err_tol) { set_err_tol(*this, err_tol); }
+
+    static void set_max_iter(Options& options, Idx max_iter) {
+        options.handle_.call_with(PGM_set_max_iter, options.get(), max_iter);
+    }
+    void set_max_iter(Idx max_iter) { set_max_iter(*this, max_iter); }
+
+    static void set_threading(Options& options, Idx threading) {
+        options.handle_.call_with(PGM_set_threading, options.get(), threading);
+    }
+    void set_threading(Idx threading) { set_threading(*this, threading); }
+
+    static void set_short_circuit_voltage_scaling(Options& options, Idx short_circuit_voltage_scaling) {
+        options.handle_.call_with(PGM_set_short_circuit_voltage_scaling, options.get(), short_circuit_voltage_scaling);
+    }
+    void set_short_circuit_voltage_scaling(Idx short_circuit_voltage_scaling) {
+        set_short_circuit_voltage_scaling(*this, short_circuit_voltage_scaling);
+    }
+
+    static void set_tap_changing_strategy(Options& options, Idx tap_changing_strategy) {
+        options.handle_.call_with(PGM_set_tap_changing_strategy, options.get(), tap_changing_strategy);
+    }
+    void set_tap_changing_strategy(Idx tap_changing_strategy) {
+        set_tap_changing_strategy(*this, tap_changing_strategy);
+    }
+
+    static void set_experimental_features(Options& options, Idx experimental_features) {
+        options.handle_.call_with(PGM_set_experimental_features, options.get(), experimental_features);
+    }
+    void set_experimental_features(Idx experimental_features) {
+        set_experimental_features(*this, experimental_features);
+    }
```

Comment:
do we want to enable fluent interface syntax here? (see https://en.wikipedia.org/wiki/Fluent_interface), i.e. `options.set_calculation_type(PGM_power_flow).set_calculation_method(PGM_newton_raphson)`

if so, all these should return `Options&`. More specifically:
* `return options` for `static`
* `return *this` for non-`static`

Reply:
Why do we need `static` functions? This is a lot boilerplate.

Reply:
We decided to implement the API such that it can be used both object based or procedural. Hence non-`static` are usually calls to the static version.

Reply:
I agree that it's too much boilerplate code, but I'm still happy that we did during the implementation process:

* We were able to find issues with the design (the design paradigm of postponing decisions as long as possible will help us in the long run)
* We can now focus on the tests first and choose one or the other based on which suits our needs best

Migrating from a dual approach to a single one is trivial

Reply:
Okay, fine with me.

Reply:
@mgovers I think I never addressed your initial question in this post. I don't think we should enable fluent interface syntax as you suggest, but rather keep it as is to be more coherent with the rest of the API.